iceoryx_posh/include/iceoryx_posh/popo/trigger_handle.hpp

```diff
@@ -86,7 +85,7 @@ class TriggerHandle
 
   private:
     ConditionVariableData* m_conditionVariableDataPtr = nullptr;
-    cxx::function<void(uint64_t)> m_resetCallback;
+    cxx::function<void(uint64_t)> m_resetCallback = [](auto) {};
```

Comment:
It wouldn't make a functional difference and you know this class better than me but would semantically an optional be better?

Just asking, since you used the optional with the ScopeGuard and you could also simplify that one with this pattern. Not sure if this is something we should suggest to use.

Reply:
Yes an optional would be much better and would lead to a huge refactoring with a lot of funny template errors ...

I would like to keep this fun for later.